frontends/web/src/api/bitbox02.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2021 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { apiGet, apiPost } from '../utils/request';
+import { TSuccess, TFail } from './response';
+
+export type TDeviceInfo = {
+    initialized: boolean;
+    mnemonicPassphraseEnabled: boolean;
+    name: string;
+    securechipModel: string;
+    version: string;
+}
+
+interface IDeviceInfo extends TSuccess {
```

Comment:
All looks great but I have one suggestion: `IDeviceInfo` is too close to `TDeviceInfo`. Easy to mistake when reading the code where these two are used. How about using a `XxxResponse` convention? IDeviceInfo would become `DeviceInfoResponse`.

Reply:
Also, this is probably beyond my comprehension but how is it that an interface "extends" from a concrete type?

Reply:
> Also, this is probably beyond my comprehension but how is it that an interface "extends" from a concrete type?

I might abuse union types here… 🤔

Reply:
@x1ddos when do we omit the `I` ?
`DeviceInfoResponse` vs `IDeviceInfoResponse`

Reply:
`TFail` becomes `FailResponse` ?

Reply:
Changed to `SuccessResponse`, `FailResponse` and started removing the `T` prefix here.

Also with the last commit we check now in the api `DeviceInfoResponse` for success and only return `DeviceInfo`.